src/plugins/mdraid.c

```diff
@@ -212,6 +214,9 @@ static BDMDExamineData* get_examine_data_from_table (GHashTable *table, gboolean
         data->size = 0;
 
     data->uuid = g_strdup ((gchar*) g_hash_table_lookup (table, "Array UUID"));
+    if (!data->uuid)
+        /* also try just "UUID" which may be reported e.g for IMSM FW RAID */
+        data->uuid = g_strdup ((gchar*) g_hash_table_lookup (table, "UUID"));
```

Comment:
This should be the first UUID (i.e. the container UUID). Looks like all the MD RAID information gathering needs to get much better, probably using attributes in /sys and+or udev data.